packages/protocol/package.json

```diff
@@ -8,7 +8,7 @@
     "build:contracts": "hardhat compile",
     "build:tsc": "rimraf -rf ./dist && yarn build:contract-dts && yarn build:fix-contract-dts && yarn build:copy-contract-dts && tsc -p tsconfig.build.json",
     "build:contract-dts": "yarn typechain --target ethers-v5 --outDir ./src/contracts './artifacts/contracts/**/!(*.dbg).json'",
-    "build:fix-contract-dts": "grep -rl 'extends ethers.utils.Interface' src/contracts | xargs sed -i 's/ethers\\.utils\\.Interface/Interface/g' && grep -rl 'Result ' src/contracts | xargs sed -i 's/Result /Result, Interface /g'",
+    "build:fix-contract-dts": "echo \"export { TypedEventFilter } from './commons'\n\" >> src/contracts/index.ts && grep -rl 'extends ethers.utils.Interface' src/contracts | xargs sed -i 's/ethers\\.utils\\.Interface/Interface/g' && grep -rl 'Result ' src/contracts | xargs sed -i 's/Result /Result, Interface /g'",
```

Comment:
Unfortunately, I had to append an export to make the TS inference work with typechain.
I'd like TypeChain to make this more developer friendly and remove this low-level handling afterwards.